admin/tool/policy/acceptances.php

```diff
@@ -28,15 +28,73 @@
 $policyid = optional_param('policyid', null, PARAM_INT);
 $versionid = optional_param('versionid', null, PARAM_INT);
 $userid = optional_param('userid', null, PARAM_INT);
+$acceptfor = optional_param('acceptfor', null, PARAM_INT);
 
-admin_externalpage_setup('tool_policy_managedocs', '', ['policyid' => $policyid, 'versionid' => $versionid],
-    new moodle_url('/admin/tool/policy/acceptances.php',
-        ['policyid' => $policyid, 'versionid' => $versionid, 'userid' => $userid]));
-require_capability('tool/policy:viewacceptances', \context_system::instance());
+if ($acceptfor) {
+    $versionid = required_param('versionid', PARAM_INT);
+    $usercontext = context_user::instance($acceptfor);
+    require_capability('tool/policy:acceptbehalf', $usercontext);
+    $userid = $acceptfor;
+}
 
-$output = $PAGE->get_renderer('tool_policy');
+// Set up the page either in an individual user context or as an admin page 'tool_policy_managedocs'.
+if ($userid || !has_capability('tool/policy:viewacceptances', context_system::instance())) {
+    // Viewing report for the individual user - either for oneself or for mentee or this is a privacy officer.
+    require_login();
+    if (isguestuser()) {
+        print_error('noguest');
+    }
+    $context = context_user::instance($userid ?: $USER->id);
+    if ($userid && $USER->id != $userid) {
+        if (!has_any_capability(['tool/policy:acceptbehalf', 'tool/policy:viewacceptances'], $context)) {
+            require_capability('tool/policy:viewacceptances', $context);
```

Comment:
I feel a bit lost here (this is where human readable inline comments help), what is the point? If the user does not have any of the listed caps, then the require_capability just throws error? What is this block really checking?

Reply:
At the moment I use the same page acceptances.php for displaying and accepting site policies (on behalf of somebody). I'm waiting for the mockups from UX team to finalise interface and it will affect validation. Most likely I'll rewrite this block anyway. But in any case I'll comment it properly.

Reply:
Thanks.